File: 1161-Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxLevelSum(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        curr_level = max_level = 1
        max_sum = float('-inf')
        
        queue = collections.deque([root])
        
        while queue:
            curr_sum = 0
            for _ in range(len(queue)):
                curr = queue.popleft()
                curr_sum += curr.val
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
            if curr_sum > max_sum:
                max_level = curr_level
                max_sum = curr_sum
            curr_level += 1
        
        return max_level

File: C++/1161-Maximum Level Sum of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int level = 1, mxVal = INT_MIN, res = 1;

        queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            int sz = q.size(), sum = 0;;
            while (sz-- > 0) {
                auto curr = q.front();
                sum += curr->val;
                q.pop();
                if (curr->left != nullptr) {
                    q.push(curr->left);
                }
                if (curr->right != nullptr) {
                    q.push(curr->right);
                }
            }
            if (sum > mxVal) {
                mxVal = sum;
                res = level;
            }
            level++;
        }

        return res;
    }
};